### PR TITLE
Initial ESM conversion

### PR DESCRIPTION
**Pull request by fgblomqvist**
Converts the server, the app entrypoint, and the controllers entrypoint to ESM. GitHub's change detection is a bit subpar so I recommend stepping through the commits to see the actual changes.

### CODE REVIEW

**Comment by fgblomqvist** [comment]
Open to questions/feedback of course!

File: docker-compose.yml
Comment (fgblomqvist) [inline docker-compose.yml:25]: Not sure if this is a miss or if I'm missing something? Couldn't get `docker-compose up` to work without it. Or perhaps the REDIS_HOST etc. should just be set under the env section?

**Review by simonv3 — APPROVED** [review]
Thanks for taking this step! 

Edit: didn't realize I'd have to approve the checks to run before they'd run.

We use [comitlint](https://commitlint.js.org/#/) for lint messages, and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to make them machine / human read friendly. Won't block the merge on account of those, but just a heads up for the future! 

Ditto we use [standard js](https://standardjs.com/) for styling the code. I just have standard run automatically on save (I'm impartial on what we use, standard was in place in all the projects, I've noticed there's some other things that would be nice to lint against that standard doesn't catch, like `only` commands in the tests. This isn't a biggy, again, whoever touches those files next time will just press save and it'll fix itself, but just a heads up again :)

Realize we don't mention either of those in the readme! So that's on us. 

The main one is that the tests pass :)

File: docker-compose.yml
Comment (simonv3) [reply docker-compose.yml:25]: Hmm, it's been working fine for me without setting it explicitly, and it looks like [that's expected](https://docs.docker.com/compose/environment-variables/#substitute-environment-variables-in-compose-files)

> You can set default values for environment variables using a [.env file](https://docs.docker.com/compose/env-file/), which Compose automatically looks for in project directory (parent folder of your Compose file). Values set in the shell environment override those set in the .env file.

I don't think referencing it explicitly is a problem?

File: src/controllers/index.mjs
Comment (simonv3) [inline src/controllers/index.mjs:8]: Not for this PR, and it was a problem beforehand, but this does feel like a very inefficient way to load everything, but while I was trying to fix the open-api thing this is what I found that actually loads everything correctly. I'm sure there's a better way, so if you think of something cleaner, all ears!

File: docker-compose.yml
Comment (fgblomqvist) [reply docker-compose.yml:25]: It was trying to connect to Redis on 127.0.0.1, which is the default when no env. var. is set. However, mystery is solved: I had installed docker-compose from the Fedora repo but that was some old legacy version. Installed the new plugin now (so that I can do `docker compose up`) and now it works as expected without specifying the env_file. That being said, part of what confused me was that the `pgsql` container had the `env_file` listed as well. Should perhaps be cleaned up as well :thinking:

Comment (fgblomqvist) [reply docker-compose.yml:25]: Jokes on me, that was too early of a celebration. It did not solve it. I tried both the plugin and the standalone, v2.11.2. Still not loading the env file automatically unless it's specified in the compose file

**Comment by fgblomqvist** [comment]
Noted re: commitlint! Should indeed be added to the README. As for standardjs, probably makes sense to switch to ESLint like the rest of the world has (and Prettier along with that if you want good automatic formatting). Either way, until then, makes sense that all the checks in an MR should pass before merging (didn't notice them since they didn't run automatically, but they should in the future now that I've had a PR merged) :slightly_smiling_face:

File: src/controllers/index.mjs
Comment (fgblomqvist) [reply src/controllers/index.mjs:8]: You mean importing all the different endpoints? Or this specific api-dock.js import? :sweat_smile:

**Comment by fgblomqvist** [comment]
Hmm should commitlint be added as a devdep to the package.json? Or how do I know which version of commitlint to use so that it aligns with what is run in the pipeline?

File: docker-compose.yml
Comment (simonv3) [reply docker-compose.yml:25]: Okay well lets leave it as is for now and see if it solves it self. 

Now that I'm thinking about it I think I actually encountered the same issue within the GitHub Action that it wasn't finding the env file in the test docker-compose. So seems safest to have it mentioned.

File: src/controllers/index.mjs
Comment (simonv3) [reply src/controllers/index.mjs:8]: oh haha all the different endpoints, not just this one.

**Comment by simonv3** [comment]
Yeah, I use prettier and eslint at work and prefer it (beam uses it too!). Would be happy to transition to it, would definitely make my life a bit easier. 

commitlint: in the action we just use the commitlint from [this GitHub Action](https://github.com/wagoid/commitlint-github-action)

File: docker-compose.yml
Comment (fgblomqvist) [reply docker-compose.yml:25]: For posterity: Found out the reason why this needs to be specified: https://stackoverflow.com/a/58048994/998919
Essentially, no env vars will be set inside the container if you don't specify the .env file. The reason why it might be working for you without setting the env vars might be because you're on a different OS that works a bit differently perhaps? Do you happen to be running macOS? I think only linux 100% "correctly" runs docker, both macOS and Windows fudges some things with the networking. Either way, if we expect the env vars to be used inside the containers, let's make sure to specify the "env_file".

Comment (fgblomqvist) [reply docker-compose.yml:25]: Clueless then. Is your api really able to connect to the db/Redis on 127.0.0.1?

**Pull request merged**
